app/src/main/java/com/fstyle/structure_android/utils/validator/Validator.java

```diff
@@ -222,41 +221,29 @@ public String getErrorByType(@ValidType int type) {
         return mContext.getString(mAllErrorMessage.keyAt(index));
     }
 
-    public void initNGWordPattern() {
-        Observable.create(new Observable.OnSubscribe<Pattern>() {
-            @Override
-            public void call(Subscriber<? super Pattern> subscriber) {
-                try {
-                    BufferedReader reader = new BufferedReader(
-                            new InputStreamReader(mContext.getAssets().open("ng-word")));
-                    StringBuffer buffer = new StringBuffer();
-                    String line;
-                    while ((line = reader.readLine()) != null) {
-                        if (buffer.length() == 0) {
-                            buffer.append("(").append(line.toLowerCase(Locale.ENGLISH));
-                        } else {
-                            buffer.append("|").append(line.toLowerCase(Locale.ENGLISH));
-                        }
+    public Subscription initNGWordPattern() {
+        return Observable.create((Observable.OnSubscribe<Pattern>) subscriber -> {
+            try {
+                BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(mContext.getAssets().open("ng-word")));
+                StringBuilder buffer = new StringBuilder();
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    if (buffer.length() == 0) {
+                        buffer.append("(").append(line.toLowerCase(Locale.ENGLISH));
+                    } else {
+                        buffer.append("|").append(line.toLowerCase(Locale.ENGLISH));
```

Comment:
WARNING: [AppendCharacterWithChar] Avoid appending characters as strings in StringBuffer.append.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AppendCharacterWithChar